lib/MusicBrainz/Server/Controller/Root.pm

```diff
@@ -154,6 +154,7 @@ sub error_404 : Private {
 
     $c->response->status(404);
     $c->stash->{current_view} = 'Node';
+    $c->stash->{component_props}->{message} = $c->stash->{message} if $c->stash->{message};
```

Comment:
Instead of having to add the message to the stash, we can accept a `$message` argument here. You can pass the message like `$c->detach('/error_404', ['message'])`. That seems easier to follow to me (because if you add something to the stash, it's hard to tell what's using it or where it's being used later).

Reply:
Done!